lessons/programming/algorithm/index.mdx

```diff
@@ -8,14 +8,69 @@ title: Algorithm
 
 In real-life if you had to count the number of members in a room, you would count one by one until you finish counting the last member.
 
-So we count a member and add a next to the previous counted number and repeat the same procedure until there are no members in the room. This is a step-wise procedure and that a normal person follows to count the members.
+So we count a member and add one to the previous counted number, go to next member, and repeat the same procedure until there are no members in the room. This is a step-wise procedure and that a normal person follows to count the members.
 
-For a computer to do the exact steps like humans, we create an algorithm containing the steps to execute. In this case, pseudo-code of an algorithm to count the members will be like.
+For a computer to solve a problem or perform a task for us, for example, a counting poblem, we give it a sequence of instructions similar to the step-bt-step procedure humans follow ourselves. An algorithm is a sequence of instructions given to computers to solve problems or perform tasks.
+
+An algorithm can be written by a plain langugae, a pseudo-code or described by a diagram-representd flowchart and it can be implemented in different computer programming langugages. 
+
+## Pseudo-code
+Pseudo-codes are simplified version of computer programs and written in nantural language.
+
+- Pseudo-code can be read and followed by programmers to write computer program.
+- Pseudo-code does not have specific syntax and can't be excuted by a computer.
+- Pseudo-code can be implemented in specific programming language.
+
+for example, a pseudo-code of counting members is as follows:
+
+num_members_far is set to be zero before counting
+while the input of member name is not empty
+    increment number_members_so_far by one
+num_members_so_far is the total number of members at the end
```

Comment:
Might be better to change the var name from **num_members_far** -> **total_members** or **num_members_count**

Then describing each pseudo variable name in a list:
 **Name:** Description
 **Name2:** Description2

**total_members**: The total number of members at the end.

_You can also utilize a feature that was recently added by @hcs847. You can **Note:** to emphasize important information such as **is set to zero before counting.** If there's more than one note or that page is long enough, the users would be able to see Note highlights at the top._
![stack](https://user-images.githubusercontent.com/33711869/120169754-e5ca1b80-c21d-11eb-903b-c0da837e9d3f.gif)
 
You may wanna use line break `<br />` tags and **bold** markdown tags since the steps are currently showing up like this: 
![Screenshot from 2021-05-31 14-43-34](https://user-images.githubusercontent.com/33711869/120170521-abad4980-c21e-11eb-855b-d729c3410193.png)